src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessorClientFactory.cs

```diff
@@ -0,0 +1,52 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Azure.Identity;
+using Azure.Messaging.EventHubs;
+using Azure.Storage.Blobs;
+using EnsureThat;
+using Microsoft.Health.Common.Auth;
+
+namespace Microsoft.Health.Events.EventHubProcessor
+{
+    public class EventProcessorClientFactory
+    {
+        private BlobContainerClient _blobContainerClient;
+
+        public EventProcessorClientFactory(BlobContainerClient blobContainerClient)
+        {
+            _blobContainerClient = EnsureArg.IsNotNull(blobContainerClient, nameof(blobContainerClient));
+        }
+
+        public EventProcessorClient CreateProcessorClient(EventProcessorClientFactoryOptions options, EventProcessorClientOptions eventProcessorClientOptions)
+        {
+            EnsureArg.IsNotNull(options);
+
+            if (options.ServiceManagedIdentityAuth)
```

Comment:
>ServiceManagedIdentityAuth [](start = 24, length = 26)

Will this be extended to support cert based auth as well for customer facing MI?

Reply:
For cert based auth I created a second method in EventProcessorClientFactory which takes an IAzureCredentialProvider which is a way to specify a custom way of retrieving a token. I was thinking of passing in a provider that does the cert based auth.

EventProcessorClient CreateProcessorClient(IAzureCredentialProvider provider, BlobContainerClient blobContainerClient, EventProcessorClientFactoryOptions options, EventProcessorClientOptions eventProcessorClientOptions)